packages/@aws-cdk/aws-apprunner-alpha/lib/auto-scaling-configuration.ts

```diff
@@ -0,0 +1,194 @@
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Imports an App Runner Auto Scaling Configuration from its ARN
+   */
+  public static fromArn(scope: Construct, id: string, autoScalingConfigurationArn: string): IAutoScalingConfiguration {
+    const arn = cdk.Stack.of(scope).splitArn(autoScalingConfigurationArn, cdk.ArnFormat.SLASH_RESOURCE_NAME);
+
+    const resourceParts = arn.resourceName?.split('/');
+
+    if (!resourceParts || resourceParts.length < 3) {
+      throw new Error(`Unexpected ARN format: ${autoScalingConfigurationArn}`);
+    }
+
+    const autoScalingConfigurationName = resourceParts[0];
+    const autoScalingConfigurationRevision = parseInt(resourceParts[1]);
+
+    return AutoScalingConfiguration.fromAutoScalingConfigurationAttributes(scope, id, {
+      autoScalingConfigurationArn,
+      autoScalingConfigurationName,
+      autoScalingConfigurationRevision,
+    });
+  }
+
+  /**
+    * The ARN of the Auto Scaling Configuration.
+    * @attribute
+    */
+  readonly autoScalingConfigurationArn: string;
+
+  /**
+    * The name of the Auto Scaling Configuration.
+    * @attribute
+    */
+  readonly autoScalingConfigurationName: string;
+
+  /**
+   * The revision of the Auto Scaling Configuration.
+   * @attribute
+   */
+  readonly autoScalingConfigurationRevision: number;
+
+  public constructor(scope: Construct, id: string, props: AutoScalingConfigurationProps = {}) {
+    super(scope, id, {
+      physicalName: props.autoScalingConfigurationName,
+    });
+
+    if (props.autoScalingConfigurationName !== undefined) {
+      if (props.autoScalingConfigurationName.length < 4 || props.autoScalingConfigurationName.length > 32) {
+        throw new Error(`autoScalingConfigurationName must be between 4 and 32 characters long, but it has ${props.autoScalingConfigurationName.length} characters.`);
+      }
+
+      if (!/^[A-Za-z0-9][A-Za-z0-9\-_]{3,31}$/.test(props.autoScalingConfigurationName)) {
+        throw new Error(`autoScalingConfigurationName ${props.autoScalingConfigurationName} must start with a letter or number, and can contain only letters, numbers, hyphens, and underscores.`);
+      }
+    }
+
+    if (props.minSize !== undefined && !cdk.Token.isUnresolved(props.minSize) && (props.minSize < 1 || props.minSize > 25)) {
+      throw new Error(`minSize must be between 1 and 25, got ${props.minSize}`);
+    }
+    if (props.maxSize !== undefined && (props.maxSize < 1 || props.maxSize > 25)) {
+      throw new Error(`maxSize must be between 1 and 25, got ${props.maxSize}`);
+    }
+    if (props.minSize !== undefined && props.maxSize !== undefined && !(props.minSize < props.maxSize)) {
+      throw new Error('maxSize must be greater than minSize');
+    }
+    if (props.maxConcurrency !== undefined && (props.maxConcurrency < 1 || props.maxConcurrency > 200)) {
+      throw new Error(`maxConcurrency must be between 1 and 200, got ${props.maxConcurrency}`);
+    }
```

Comment:
```suggestion
    const isMinSizeDefined = props.minSize !== undefined && !cdk.Token.isUnresolved(props.minSize)
    const isMaxSizeDefined = props.maxSize !== undefined && !cdk.Token.isUnresolved(props.maxSize)
    if (isMinSizeDefined && (props.minSize < 1 || props.minSize > 25)) {
      throw new Error(`minSize must be between 1 and 25, got ${props.minSize}`);
    }
    if (isMaxSizeDefined && (props.maxSize < 1 || props.maxSize > 25)) {
      throw new Error(`maxSize must be between 1 and 25, got ${props.maxSize}`);
    }
    if (isMinSizeDefined && isMaxSizeDefined && !(props.minSize < props.maxSize)) {
      throw new Error('maxSize must be greater than minSize');
    }
    if (
      props.maxConcurrency !== undefined && 
      !cdk.Token.isUnresolved(props.maxConcurrency) &&
      (props.maxConcurrency < 1 || props.maxConcurrency > 200)
    ) {
      throw new Error(`maxConcurrency must be between 1 and 200, got ${props.maxConcurrency}`);
    }
```
We should check if resolved for all validated props